sim-cli/src/parsing.rs

```diff
@@ -554,7 +567,11 @@ pub async fn get_validated_activities(
     clients: &HashMap<PublicKey, Arc<Mutex<dyn LightningNode>>>,
     nodes_info: HashMap<PublicKey, NodeInfo>,
     activity: Vec<ActivityParser>,
-) -> Result<Vec<ActivityDefinition>, LightningError> {
+) -> Result<ActivityOpts, LightningError> {
+    if activity.is_empty() {
```

Comment:
It's odd to me to split random activity `ActivityOpts` between here and the call site - either this function is responsible for assembling all of the possible `ActivityOpts` or it's just for defined (non-random) activities, now sure what the motivation to split it like this is?